debugtools/DDR_VM/src/com/ibm/j9ddr/corereaders/memory/PatchedMemoryList.java

```diff
@@ -0,0 +1,393 @@
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package com.ibm.j9ddr.corereaders.memory;
+
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class statically maintains a list of memory records are in ascending order, meaning the end address of the previous record is 
+ * always less than the start address of the next address.  If the end address of the previous record is equal to the start
+ * address of the next address, then two records will be merged to one single record.
+ * </P>
+ * This class provides two major static functions:
+ * (1) Add a memory record as a patch.  Note, the memory content in the new memory record will overwrite the memory content in the existing list
+ * if they are for the same address.
+ * (2) Patch the memory from a specified area. Note, the result is a list of memory records.  If a memory record has only the start address and the length,
+ * but the content of the memory record is null, then the memory record is not patched.  Otherwise, the memory record has been patched.
+ * <p>
+ * @author Manqing LI, IBM Canada
+ *
+ */
+public class PatchedMemoryList {
+	public static boolean verbose = new Boolean(System.getProperty("com.ibm.j9ddr.corereaders.memory.PatchedMemoryList.verbose"));
+	public static List<MemoryRecord> list = null;  // a list with start address in ascending order; no memory content between two elements overlaps.
+
+	/**
+	 * Record a new memory record.
+	 * <p>
+	 * @param important	The new memory record whose content will be used to overwrite the existing content in the list.
+	 */
+	public static void record(MemoryRecord important) {
+		addAndMergeIfNecessary(important);
+		if (verbose) {
+			PatchedMemoryList.print(System.out);
+		}
+	}
+
+	/**
+	 * Add the new record to the list and merge if necessary.
+	 * <p>
+	 * @param important	The new memory record whose content will be used to overwrite the existing content in the list.
+	 */
+	private static void addAndMergeIfNecessary(MemoryRecord important) {
+		if (list == null) {
+			list = new ArrayList<MemoryRecord>();
+		}
+		for (int index = 0; index < list.size(); index++) {
+			if (important.getStart() > list.get(index).getEnd()) {
+				continue; // the new one is separately higher than the current one, go to the next one.
+			} else if (important.getEnd() < list.get(index).getStart()){
```

Comment:
Some local variables are in order here, for `important.getStart()`, `important.getEnd()`, `list.size()`, `list.get(index)`, etc.